secp256k1-sys/src/lib.rs

```diff
@@ -258,239 +300,128 @@ impl hash::Hash for KeyPair {
                                        out: *mut PublicKey,
                                        ins: *const *const PublicKey,
                                        n: c_int)
                                        -> c_int;
 
-    #[cfg_attr(not(rust_secp_no_symbol_renaming), link_name = "rustsecp256k1_v0_4_0_ecdh")]
-    pub fn secp256k1_ecdh(
-        cx: *const Context,
-        output: *mut c_uchar,
-        pubkey: *const PublicKey,
-        seckey: *const c_uchar,
-        hashfp: EcdhHashFn,
-        data: *mut c_void,
-    ) -> c_int;
-
-    // Extra keys
-    #[cfg_attr(not(rust_secp_no_symbol_renaming), link_name = "rustsecp256k1_v0_4_0_keypair_create")]
-    pub fn secp256k1_keypair_create(
-        cx: *const Context,
-        keypair: *mut KeyPair,
-        seckey: *const c_uchar,
-    ) -> c_int;
-
-    #[cfg_attr(not(rust_secp_no_symbol_renaming), link_name = "rustsecp256k1_v0_4_0_xonly_pubkey_parse")]
-    pub fn secp256k1_xonly_pubkey_parse(
-        cx: *const Context,
-        pubkey: *mut XOnlyPublicKey,
-        input32: *const c_uchar,
-    ) -> c_int;
-
-    #[cfg_attr(not(rust_secp_no_symbol_renaming), link_name = "rustsecp256k1_v0_4_0_xonly_pubkey_serialize")]
-    pub fn secp256k1_xonly_pubkey_serialize(
-        cx: *const Context,
-        output32: *mut c_uchar,
-        pubkey: *const XOnlyPublicKey,
-    ) -> c_int;
-
-    #[cfg_attr(not(rust_secp_no_symbol_renaming), link_name = "rustsecp256k1_v0_4_0_xonly_pubkey_from_pubkey")]
-    pub fn secp256k1_xonly_pubkey_from_pubkey(
-        cx: *const Context,
-        xonly_pubkey: *mut XOnlyPublicKey,
-        pk_parity: *mut c_int,
-        pubkey: *const PublicKey,
-    ) -> c_int;
-
-    #[cfg_attr(not(rust_secp_no_symbol_renaming), link_name = "rustsecp256k1_v0_4_0_xonly_pubkey_tweak_add")]
-    pub fn secp256k1_xonly_pubkey_tweak_add(
-        cx: *const Context,
-        output_pubkey: *mut PublicKey,
-        internal_pubkey: *const XOnlyPublicKey,
-        tweak32: *const c_uchar,
-    ) -> c_int;
-
-    #[cfg_attr(not(rust_secp_no_symbol_renaming), link_name = "rustsecp256k1_v0_4_0_keypair_xonly_pub")]
-    pub fn secp256k1_keypair_xonly_pub(
-        cx: *const Context,
-        pubkey: *mut XOnlyPublicKey,
-        pk_parity: *mut c_int,
-        keypair: *const KeyPair
-    ) -> c_int;
-
-    #[cfg_attr(not(rust_secp_no_symbol_renaming), link_name = "rustsecp256k1_v0_4_0_keypair_xonly_tweak_add")]
-    pub fn secp256k1_keypair_xonly_tweak_add(
-        cx: *const Context,
-        keypair: *mut KeyPair,
-        tweak32: *const c_uchar,
-    ) -> c_int;
-
-    #[cfg_attr(not(rust_secp_no_symbol_renaming), link_name = "rustsecp256k1_v0_4_0_xonly_pubkey_tweak_add_check")]
-    pub fn secp256k1_xonly_pubkey_tweak_add_check(
-        cx: *const Context,
-        tweaked_pubkey32: *const c_uchar,
-        tweaked_pubkey_parity: c_int,
-        internal_pubkey: *const XOnlyPublicKey,
-        tweak32: *const c_uchar,
-    ) -> c_int;
```

Comment:
Are these unused in Grin?

Reply:
All of these functions does not yet exist in https://github.com/mimblewimble/secp256k1-zkp

Reply:
Can we leave these in and stub them out (for now) in the lib? This would allow us to remain closer to upstream here.